### PR TITLE
Make KVM support optional

### PR DESCRIPTION
**Pull request by jakogut**
One think I'd like to look into further is if the Makefile `$(COMPOSEBIN)` target is getting executed even when the target file already exists, as it prints a message every time.

### CODE REVIEW

**Comment by klutchell** [comment]
@jakogut the COMPOSEBIN stuff was reworked in the first 3 commits of https://github.com/balena-os/leviathan/pull/568. Might want to give it a try, just cherry-pick the commits in question and we can break them into a separate PR if needed.

File: worker/Dockerfile.template [hunk 1]
```diff
@@ -47,4 +47,4 @@ COPY --from=node-build /tmp/node/package.json .
 COPY --from=node-build /tmp/node/node_modules node_modules
 COPY --from=node-build /tmp/node/build build
 
-CMD [ "./entry.sh" ]
+CMD ./entry.sh
```
Comment (klutchell) [inline worker/Dockerfile.template:50]: y tho? the former is the preferred syntax

File: Makefile [hunk 2]
```diff
@@ -22,12 +22,8 @@ LOCALCOMPOSEFILE := ./docker-compose.local.yml
 
 # create a docker-compose binary either as a link or a script
 $(COMPOSEBIN):
-ifneq ($(shell command -v docker-compose 2>/dev/null),)
-	ln -sf $(shell command -v docker-compose 2>/dev/null) $@
-else
 	curl -fsSL "https://github.com/docker/compose/releases/download/1.29.2/run.sh" -o $@
 	chmod +x $@
-endif
```
Comment (klutchell) [inline Makefile:30]: Did you mean to commit this? I haven't tested myself but I assume most docker-compose versions will work?

File: worker/Dockerfile.template (hunk 1 above)
Comment (jakogut) [reply worker/Dockerfile.template:50]: @klutchell This is to execute the command with a shell, which results in our messages in the entry script being printed to stdout. With the JSON array format, the script is executed without a shell, and the messages aren't printed.

File: Makefile (hunk 2 above)
Comment (jakogut) [reply Makefile:30]: Yes, it was intentional, check out the commit message.

File: worker/entry.sh [hunk 3]
```diff
@@ -1,2 +1,25 @@
...
+if lsmod | grep '\<tun\>' > /dev/null; then
+	tun_minor=$(grep tun /proc/misc | cut -d ' ' -f1)
+	mkdir -p /dev/net
+	if ! mknod -m 666 /dev/net/tun c 10 "${tun_minor}"; then
+		echo "Unable to create TUN device node"
+		exit 1
+	fi
+else
+	echo "TUN module not loaded, unable to setup networking"
+	exit 1
```
Comment (klutchell) [inline worker/entry.sh:12]: I want to approve this for the kvm stuff but now I'm missing tun networking, so I'm glad you added this check.
```
worker_1  | TUN module not loaded, unable to setup networking
leviathan-generic-test_worker_1 exited with code 1
```
When I get a chance I'll try to enable the tun module on an ec2 spot instance and hopefully approve this PR.

**Comment by jakogut** [comment]
@balena-ci rebase

File: worker/entry.sh (hunk 3 above)
Comment (jakogut) [reply worker/entry.sh:12]: @klutchell I've received this same error running the worker after a fresh boot, try loading the module before running the worker with `modprobe tun`.

Comment (klutchell) [reply worker/entry.sh:12]: It seems that `lsmod | grep '\<tun\>'` will fail if the module is builtin.
```
modprobe tun
```
```
lsmod | grep tun
```
```
modinfo tun 
name:           tun
filename:       (builtin)
alias:          devname:net/tun
alias:          char-major-10-200
license:        GPL
file:           drivers/net/tun
author:         (C) 1999-2004 Max Krasnyansky <maxk@qualcomm.com>
description:    Universal TUN/TAP device driver
```

**Review by klutchell — APPROVED** [review]
lgtm

**Pull request merged**